rpcserver.go

```diff
@@ -1103,3 +1108,80 @@ func (r *rpcServer) SendAsset(ctx context.Context,
 		TotalFeeSats: int64(resp.TotalFees),
 	}, nil
 }
+
+// SubscribeSendAssetEventNtfns registers a subscription to the event
+// notification stream which relates to the asset sending process.
+func (r *rpcServer) SubscribeSendAssetEventNtfns(
+	in *tarorpc.SubscribeSendAssetEventNtfnsRequest,
+	ntfnStream tarorpc.Taro_SubscribeSendAssetEventNtfnsServer) error {
+
+	eventSubscriber := chanutils.NewEventReceiver[tarofreighter.Event](
+		chanutils.DefaultQueueSize,
+	)
+	err := r.cfg.ChainPorter.RegisterSubscriber(eventSubscriber, false, false)
+	if err != nil {
+		return err
+	}
+
+	// This goroutine will read from the ChainPorter event subscription and
+	// forward to the RPC stream.
+	errorsChan := make(chan error, 1)
+	go func() {
+		for {
+			select {
+			case event := <-eventSubscriber.NewItemCreated.ChanOut():
+
+				rpcEvent, err := marshallSendAssetEvent(event)
+				if err != nil {
+					err := fmt.Errorf("failed to marshall"+
+						"ChainPorter event into RPC "+
+						"event: %w", err)
+					errorsChan <- err
+					return
+				}
+
+				err = ntfnStream.Send(rpcEvent)
+				if err != nil {
+					err := fmt.Errorf("failed to RPC "+
+						"stream send event: %w", err)
+					errorsChan <- err
+					return
+				}
+
+			case <-r.quit:
+				return
+			}
+		}
+	}()
+
+	select {
+	// Handle errors from the goroutine.
+	case err := <-errorsChan:
+		if err != nil {
+			return fmt.Errorf("error forwarding send asset event "+
+				"notifications to RPC stream send: %w", err)
+		}
+	case <-r.quit:
+		return nil
+	}
+
+	return nil
+}
+
+// marshallSendAssetEvent maps a ChainPorter event to its RPC counterpart.
+func marshallSendAssetEvent(eventInterface tarofreighter.Event) (*tarorpc.SendAssetEvent, error) {
+	switch event := eventInterface.(type) {
+	case *tarofreighter.ExecuteSendStateEvent:
+		eventRpc := tarorpc.SendAssetEvent_ExecuteSendStateEvent{
+			ExecuteSendStateEvent: &tarorpc.ExecuteSendStateEvent{
+				Timestamp: event.Timestamp().Unix(),
+				SendState: event.SendState.String(),
```

Comment:
What Laolu probably meant in a previous comment about an enum is to not use the `String()` on `SendState` here but instead translate it into a gRPC enum with the same elements. That makes it less fragile when it comes to renaming the states. But it also means more work to marshal/unmarshal between the different enums. So IMO not blocking for the moment (and also I'm going to change a lot of those states in my upcoming PR :sweat_smile: ).

Reply:
Ok, I'll leave as is for now.